lib/charms/opensearch/v0/opensearch_peer_clusters.py

```diff
@@ -196,6 +197,7 @@ def _new_cluster_setup(self, config: PeerClusterConfig) -> DeploymentDescription
                 pending_directives=directives,
                 typ=self._deployment_type(config, start_mode),
                 state=deployment_state,
+                profile=self._charm.performance_profile.current.typ.value,
```

Comment:
@reneradoi to leave it here what we've discussed in the chat: I have a ticket to go over our docs and make sure that specially smaller setups set the profile correctly.

Reply:
@reneradoi thanks for the pointers. Indeed there was a misplaced config set at `_on_config_changed`. I fixed it. Now, there are two other things:
1) there is one extra issue with `YamlConfSetter.replace`. I've separated it from this PR and merged the fix back here, original fix: [here](https://github.com/canonical/opensearch-operator/pull/484)
2) The `DeploymentDescription` will not get updated, *unless* there is a peer-cluster-orchestrator requirer. The reason is because I have a gateway to the `refresh_relation_data`.

Reply:
@Mehdi-Bendriss I'd rather do it on a follow-up PR

Reply:
> it feels like a gap in our config design for large deployments

Not really. 
Config design in large deployments is handled centrally in the `run` method as described above in: `deployment_desc.config` - it manages 3 config options, and the logic is reactive to the changes in "`new_cluster_setup / existing_cluster_setup`". Ran through the single point of entry: `charm._on_config_changed`

In any way. We can talk about improving the overall config management later.

Reply:
Yes, it is an interesting topic to discuss next week.